source/addonHandler/__init__.py

```diff
@@ -649,23 +627,38 @@ class AddonManifest(ConfigObj):
 # NVDA Add-on Manifest configuration specification
 # Add-on unique name
 name = string()
+
 # short  summary (label) of the add-on to show to users.
 summary = string()
+
 # Long description with further information and instructions
 description = string(default=None)
+
 # Name of the author or entity that created the add-on
 author = string()
+
 # Version of the add-on. Should preferably in some standard format such as x.y.z
 version = string()
-# The minimum required NVDA version for this add-on to work correctly
-minimumNVDAVersion = string(default=None)
-# The last NVDA version this add-on has been tested with.
-lastTestedNVDAVersion = string(default=None)
+
+# The minimum required NVDA version for this add-on to work correctly.
+# Should be less than or equal to lastTestedNVDAVersion
+minimumNVDAVersion = apiVersion(default="0.0.0")
+
+# Must be greater than or equal to minimumNVDAVersion
+lastTestedNVDAVersion = apiVersion(default="0.0.0")
+
 # URL for more information about the add-on. New versions and such.
 url= string(default=None)
+
 # Name of default documentation file for the add-on.
 docFileName = string(default=None)
 
+# NOTE: apiVersion:
+# Eg: 2019.1.0 or 0.0.0
+# Must have 3 integers separated by dots.

```

Comment:
Dev guide, I can fix that!

> unnecessary extra complexity?

Possibly, but I think it's a trade for removing ambiguity. To make sure I'm on the same page, do you think you could explain the complexity that you are concerned about?

Consider the following options:
- Allow either 2019.1.0 or 2019.1
  - We must assume that the minor part is zero when it's missing.
  - We will probably have to detect this and fill it in during manifest validation
- Allow only 3 part (2019.1.0) as it is now
  - We have to validate the addon manifest for 3 parts
- Allow only 2 part (2019.1)
  - We still have to validate the addon manifest for 2 parts.
  - We lose the possibility of changing the addon api version in a minor release. It's unlikely that we would ever want to do this, but the possibility is still there.

Reply:
> I could think of a case where an author marks his add-on compatible with NVDA API Version 2018.4, which would effectively make his add-on fail to load because 2018.4 translates to unknown, since it's missing the minor part.

In this case they will get a validation error. The version strings will be in the log. I should point out that any add-ons that have been updated to work with the first version of the add-on compatibility check, likely used a 2 part version string. These will need to be updated and re-released. This, I think, has the benefit that the add-on authors will reconsider how the compatibility system works and hopefully reduce errors based on assumptions made against the last implementation.